lib/Sema/CSApply.cpp

```diff
@@ -247,8 +247,16 @@ getImplicitMemberReferenceAccessSemantics(Expr *base, VarDecl *member,
     }
   }
 
+  // Check whether this is a member access on 'self'.
+  bool isAccessOnSelf = false;
+  if (auto *baseDRE = dyn_cast<DeclRefExpr>(base->getValueProvidingExpr()))
+    if (auto *AFD = dyn_cast<AbstractFunctionDecl>(DC))
+      if (auto *selfDecl = AFD->getImplicitSelfDecl())
+        if (baseDRE->getDecl() == selfDecl)
```

Comment:
Please test against `VarDecl::isSelfParameter` instead. That handles the use of `self` in non-function contexts like initial value expressions.